File: src/main/java/com/gitlab/daring/image/command/parameter/FileParam.java
package com.gitlab.daring.image.command.parameter;

import java.io.File;

import static java.lang.Integer.parseInt;

public class FileParam extends StringParam {

    public int maxCount = 1;

    public FileParam(String sv) {
        super(sv);
        parseSpec();
    }

    @Override
    protected String[] parseArgs(String sv) {
        boolean isFile = new File(sv).exists();
        return isFile ? new String[]{sv} : super.parseArgs(sv);
    }

    private void parseSpec() {
        if (spec.isEmpty()) return;
        maxCount = parseInt(spec);
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/parameter/FileParam.kt
package com.gitlab.daring.image.command.parameter

import java.io.File

class FileParam(sv: String) : StringParam(sv) {

    var maxCount = 1

    init {
        if (spec.isNotEmpty()) maxCount = spec.toInt()
    }

    override fun parseArgs(sv: String): List<String> {
        val isFile = File(sv).exists()
        return if (isFile) listOf(sv) else super.parseArgs(sv)
    }

}